test/python/quantum_info/test_synthesis.py

```diff
@@ -569,6 +572,39 @@ def test_seed_289(self):
         unitary = random_unitary(4, seed=289)
         self.check_exact_decomposition(unitary.data, two_qubit_cnot_decompose)
 
+    def test_euler_basis_selection(self):
+        """Verify decomposition uses euler_basis for 1q gates."""
+
+        euler_bases = [
+            ('U3', ['u3']),
+            ('U1X', ['u1', 'rx']),
+            ('RR', ['r']),
+            ('ZYZ', ['rz', 'ry']),
+            ('ZXZ', ['rz', 'rx']),
+            ('XYX', ['rx', 'ry']),
+        ]
+
+        kak_gates = [
+            (CXGate(), 'cx'),
+            (CZGate(), 'cz'),
+            (iSwapGate(), 'iswap'),
+            (RXXGate(np.pi/2), 'rxx'),
+        ]
+
+        for basis in product(euler_bases, kak_gates):
+            (euler_basis, oneq_gates), (kak_gate, kak_gate_name) = basis
+
+            with self.subTest(euler_basis=euler_basis, kak_gate=kak_gate):
```

Comment:
should these be ddt?

Reply:
I don't have a preference, I just kept the style of the existing synthesis tests.

Reply:
I don't mind either way